### PR TITLE
4 reset state

### PR DESCRIPTION
**Pull request by pszymczyk**

### CODE REVIEW

File: src/main/groovy/com/pszymczyk/consul/ConsulProcess.groovy [hunk 1]
```diff
@@ -1,22 +1,37 @@
...
 
     private static final Logger logger = LoggerFactory.getLogger(ConsulProcess.class);
 
-    Path dataDir;
-    int httpPort;
-    Process process;
+    private Path dataDir
+    private int httpPort
+    private Process process
+    private SimpleConsulClient simpleConsulClient
```
Comment (janisz) [inline src/main/groovy/com/pszymczyk/consul/ConsulProcess.groovy]: Maybe declare this fields `final`

File: src/main/groovy/com/pszymczyk/consul/ConsulProcess.groovy [hunk 2]
```diff
@@ -1,22 +1,37 @@
...
+        this.dataDir = dataDir
+        this.httpPort = httpPort
+        this.process = process
+        this.simpleConsulClient = new SimpleConsulClient(httpPort: httpPort)
+    }
+    /**
+     * deregister all services except consul
+     */
+    void reset() {
+        simpleConsulClient.getServicesIds().each { it -> simpleConsulClient.deregister(it) }
```
Comment (janisz) [inline src/main/groovy/com/pszymczyk/consul/ConsulProcess.groovy]: You can clean KV store too

Comment (pszymczyk) [inline src/main/groovy/com/pszymczyk/consul/ConsulProcess.groovy]: Yeah I thought about it but it's hard to list all persisted entries.

Comment (janisz) [inline src/main/groovy/com/pszymczyk/consul/ConsulProcess.groovy]: No it isn't. 
To list all values use `curl http://localhost:8500/v1/kv/?recurse`
To delete all values use `curl -X DELETE http://localhost:8500/v1/kv/?recurse`

Comment (pszymczyk) [inline src/main/groovy/com/pszymczyk/consul/ConsulProcess.groovy]: Ok I will add it in another PR

**Pull request merged**